### ISSUE TITLE
BufferGeometry computeBoundingSphere Warning

### ISSUE THREAD

**Issue by Hexxonite** [mrdoob/three.js#19735]
### Hi there,

I am currently using WebWorkers to generate large `Float32Array`'s (size about 128000) and pass them back to the DOM-thread where they are sliced into smaller pieces and set in several different position-attribute-buffers.

I only need Two-Dimensional coordiinates **X and Y** for this and **Z is always 0**.

So I created the attribute liike this:

    // position attribute (2 vertices per point)
    geometry.setAttribute('position', new THREE.BufferAttribute(new Float32Array(mySize * 2), 2));

instead of this:

    // position attribute (3 vertices per point)
    geometry.setAttribute('position', new THREE.BufferAttribute(new Float32Array(mySize * 3), 3));

I did this in order to save some overhead (Z) when passing the data from the worker.
Otherwise I would have to process the array manually, which is not really efficient.

Using 2 vertices per point seems to work as intended, **BUT** in my case the console it's printing more than 200 of these INFO messages:

    [0625/160435.634:INFO:CONSOLE(541)] "THREE.BufferGeometry.computeBoundingSphere(): Computed radius is NaN. The "position" attribute is likely to have NaN values.", source: file:///D:/#####/js/three.min.js (541)

Since this is blowing up the log files, I didn't find a way to disable it and it seems to work as intended otherwise, I was wondering if this message could be removed.

Here is a short fiddle of this:    https://jsfiddle.net/th2zvr9p/

Notice how it works the same for either "dimensions" 2 or 3, but it prints the error when using dimension 2.

This might be related?    https://github.com/mrdoob/three.js/pull/13713

I'd be glad to provide more help and infos :)

##### Three.js version

- [ ] Dev
- [x] Latest
- [ ] ...

##### Browser

- [ ] All of them
- [x] Chrome (& Engine Framework), but probably all of them are affected?
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [ ] All of them
- [x] Windows
- [ ] macOS
- [ ] Linux
- [ ] Android
- [ ] iOS

##### Hardware Requirements (graphics card, VR Device, ...)

Got a 1080ti, but dont think that matters.

**Comment by Mugen87** [comment]
Duplicate of #6288.

**Comment by Hexxonite** [comment]
Sorry, my bad.

>  it's better to let the user compute the boundingSphere on their own

could you give me a small hint regarding this? I guess I don't really need this function, so could I just **null** it out e.g.: `geometry.attributes.position.computeBoundingSphere = null;` ?

Thanks in advance.

**Comment by Mugen87** [comment]
`three.js` only supports vertex data in 3D space. Meaning the `position` attribute always needs an `itemSize` of `3`.

Besides, I don't think that your proposed change works since `BufferAttribute` does not have a `computeBoundingSphere()` method. If the bounding volumes of a geometry can't be computed correctly, engine features like view frustum culling don't work.

**Comment by Hexxonite** [comment]
You seem to be right. I dug a bit deeper and as it turns out, the problem seems to originate from the `BufferAttribute` returning `NaN `values for `.getZ()` when the array doesnt use 3 vertices per point.

This is probably intended behaviour, but I found a simple fix for it, by returning ` || 0` as a default, if there was an issue resolving the array position. I run the following code once, after the window finished loading and before I create my three.js context:

`THREE.BufferAttribute.prototype.getX = function (index) { return this.array[index * this.itemSize] || 0; };
        THREE.BufferAttribute.prototype.getY = function (index) { return this.array[index * this.itemSize + 1] || 0; };
        THREE.BufferAttribute.prototype.getZ = function (index) { return this.array[index * this.itemSize + 2] || 0; };
        THREE.BufferAttribute.prototype.getW = function (index) { return this.array[index * this.itemSize + 3] || 0; };

THREE.InterleavedBufferAttribute.prototype.getX = function (index) { return this.data.array[index * this.data.stride + this.offset] || 0; };
THREE.InterleavedBufferAttribute.prototype.getY = function (index) { return this.data.array[index * this.data.stride + this.offset + 1] || 0; };
THREE.InterleavedBufferAttribute.prototype.getZ = function (index) { return this.data.array[index * this.data.stride + this.offset + 2] || 0; };
THREE.InterleavedBufferAttribute.prototype.getW = function (index) { return this.data.array[index * this.data.stride + this.offset + 3] || 0; };
`

I figured I might as well do this for all vertices, not only Z aswell as for the `InterleavedBufferAttribute` class, and so far there seems to be no issue with this behaviour.

Maybe this will help someone in the future :)
Cheers
